patchwork-gui/src/main/java/com/patchworkmc/mixin/gui/MixinKeyboard.java

```diff
@@ -74,20 +75,26 @@ private void postKeyEvent(int i, boolean[] bls, ParentElement element, int key,
 	}
 
 	@Inject(method = "method_1458", at = @At("HEAD"), cancellable = true)
-	private static void charEvent(Element element, int character, int mods, CallbackInfo info) {
+	private static void preCharTyped(Element element, int character, int mods, CallbackInfo info) {
 		if (MinecraftForge.EVENT_BUS.post(new GuiScreenEvent.KeyboardCharTypedEvent.Pre((Screen) element, (char) character, mods))) {
 			info.cancel();
-		} else if (!element.charTyped((char) character, mods)) {
-			MinecraftForge.EVENT_BUS.post(new GuiScreenEvent.KeyboardCharTypedEvent.Post((Screen) element, (char) character, mods));
 		}
 	}
 
+	@Redirect(method = "method_1458", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/Element;charTyped(CI)Z"))
+	private static boolean charTyped(Element element, char character, int mods) {
+		return element.charTyped(character, mods) || MinecraftForge.EVENT_BUS.post(new GuiScreenEvent.KeyboardCharTypedEvent.Post((Screen) element, character, mods));
```

Comment:
won't this only fire the event if element.charTyped() returns false? Is that intentional?